src/anyp/PortCfg.cc

```diff
@@ -25,13 +29,40 @@ AnyP::PortCfgPointer FtpPortList;
 int NHttpSockets = 0;
 int HttpSockets[MAXTCPLISTENPORTS];
 
-AnyP::PortCfg::PortCfg(const TrafficModeFlags::PortKind aPortKind):
+static AnyP::TrafficModeFlags::PortKind
+ParseDirective(const SBuf &directive)
+{
+    if (directive.cmp("http_port") == 0)
+        return AnyP::TrafficModeFlags::httpPort;
+    else if (directive.cmp("https_port") == 0)
+        return AnyP::TrafficModeFlags::httpsPort;
+    else {
+        assert(directive.cmp("ftp_port") == 0);
+        return AnyP::TrafficModeFlags::ftpPort;
+    }
+}
+
+static AnyP::ProtocolVersion
+DefaultProtocol(const SBuf &directive)
```

Comment:
Alternatively, we can pass here AnyP::TrafficModeFlags (which is already parsed by PortCfg constructor). That would eliminate duplicating strings analysing.

Reply:
I agree that we should get rid of one set of string comparisons. For better code readability, I believe you would have to reorder the data members, initializing `flags` using PortKind(directiveName) and then initialize `transport` using DefaultTransport(flags.portKind()).

The opposite/current initialization order would require[^1] a DefaultProtocolToPortKind() function (that accepts the Anyp::ProtocolType argument and assumes it is the _default_ port protocol) which is more difficult to swallow than simpler/straightforward PortKind() and DefaultTransport() names.

[^1]: If we name that function PortKind(), then we would be incorrectly implying that the port kind can be computed from any port protocol (even though our code would not use that ability, we would be providing it). I suspect that the current `protocol=` handling code allows bad combinations (e.g., `http_port protocol=FTP`). I would very much prefer to avoid getting these PRs into those old `protocol=` handling problems.

Reply:
Done (e60cd4).